### PR TITLE
[API] Use serializer instead of custom view for shipping methods

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | replaces https://github.com/Sylius/Sylius/pull/12392
| License         | MIT

### CODE REVIEW

**Review by Zales0123 — COMMENTED** [review]
A lot of code removed is always nice to see 🚀

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ShippingMethod.xml [hunk 1]
```diff
@@ -23,7 +23,7 @@
         </attribute>
 
         <attribute name="normalization_context">
-            <attribute name="groups">admin:shipping_method:read</attribute>
+            <attribute name="groups">shop:shipping_method:read</attribute>
```
Comment (arti0090) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/ShippingMethod.xml:26]: Wouldn't this lead to show shop normalization for admin_put, delete and archive? Maybe deleting this block would be better and declare normalization context where it is needed?

File: src/Sylius/Bundle/ApiBundle/Swagger/ShippingMethodDocumentationNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
```
Comment (arti0090) [inline src/Sylius/Bundle/ApiBundle/Swagger/ShippingMethodDocumentationNormalizer.php:2]: Doc block? Is it needed or not here?

File: src/Sylius/Bundle/ApiBundle/Swagger/ShippingMethodDocumentationNormalizer.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+{
+    /** @var NormalizerInterface */
+    private $decoratedNormalizer;
+
+    public function __construct(NormalizerInterface $decoratedNormalizer)
+    {
+        $this->decoratedNormalizer = $decoratedNormalizer;
+    }
+
+    public function supportsNormalization($data, $format = null)
```
Comment (arti0090) [inline src/Sylius/Bundle/ApiBundle/Swagger/ShippingMethodDocumentationNormalizer.php:20]:
```suggestion
    public function supportsNormalization($data, $format = null): bool 
```
🙃

**Review by Zales0123 — APPROVED** [review]

**Comment by Zales0123** [comment]
Thank you, Grzegorz! :tada:

**Pull request merged**